Documentation/README_Solver.md

```diff
@@ -1,40 +1,155 @@
+* [`InBetween`](xref:Microsoft.MixedReality.Toolkit.Utilities.Solvers.InBetween): Keeps an object in between two tracked objects.
+
+In order to use the Solver system, simply add one of the components listed above to a GameObject. Since all Solvers require a [`SolverHandler`](xref:Microsoft.MixedReality.Toolkit.Utilities.Solvers.SolverHandler), one will be created automatically by Unity.
+
+> [!NOTE]
+> Examples of how to use the Solvers system can be found in the **SolverExamples.scene** file.
+
+## How to change tracking reference
+
+The *Tracked Target Type* property of the [`SolverHandler`](xref:Microsoft.MixedReality.Toolkit.Utilities.Solvers.SolverHandler) component defines the point of reference all solvers will use to calculate their algorithms. For example, a value type of [`Head`](xref:Microsoft.MixedReality.Toolkit.Utilities.TrackedObjectType.Head) with a simple [`SurfaceMagnetism`](xref:Microsoft.MixedReality.Toolkit.Utilities.Solvers.SurfaceMagnetism) component will result in a raycast from the head and in the direction of the user's gaze for solving what surface is hit. Potential values for the `TrackedTargetType` property are:
+
+* *Head* : Point of reference from the camera
+* *MotionController*: Point of reference from a motion controller. Use `TrackedHandedness` property to define details
+* *HandJoint*: Point of reference from a hand. Use `TrackedHandJoint` and `TrackedHandedness` properties to define details
+* *CustomOverride*: Point of reference from the assigned `TransformOverride`
+
+> [!NOTE]
+> For both *MotionController* and *HandJoint* types, the solver handler will attempt to provide the left controller/hand transform first and then the right if the former is not available or unless the `TrackedHandedness` property specifies otherwise.
+
+![Solver](../Documentation/Images/Solver/TrackedObjectType-Example.gif)
+<br/>
+*Example of various properties associated with each TrackedTargetType*
+
+## How to chain Solvers
+
+It is possible to add multiple `Solver` components to the same GameObject thus chaining their algorithms. The `SolverHandler` components handles updating all solvers on the same GameObject. By default the `SolverHandler` calls `GetComponents<Solver>()` on Start which will return the Solvers in the order that they appear in the inspector.
+
+Furthermore, setting the *Updated Linked Transform* property to true will instruct that `Solver` to save it's calculated position, orientation, & scale to an intermediary variable accessible by all Solvers (i.e `GoalPosition`). When false, the `Solver` will update the GameObject's transform directly. By saving the transform properties to an intermediary location, other Solvers are able to perform their calculations starting from the intermediary variable. This is because Unity does not allow updates to gameObject.transform to stack within the same frame.
```

Comment:
I wonder if this property should have been named "allow multiple solvers" or something similar to that - I guess this is probably a discussion for a future day, but it's interesting in that, from a user-API perspective, it feels like the property should be named according to intent for what the user is trying to do, but it's more named from an implementation perspective (i.e. it does what it says, but missing is the overall intent)

Reply:
I agree. Even "Is Chained Solver" maybe? On the fence about renaming. We could also just update name in inspector but not code

Reply:
There is already a lot of changes here. Let's leave this out for now